### PR TITLE
refactor: support props

### PR DESCRIPTION
**Pull request by Sepush**
Signed-off-by: Sepush <sepush@outlook.com>

### CODE REVIEW

File: src/.vitepress/theme/index.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { EnhanceAppContext } from 'vitepress'
 import DefaultTheme from 'vitepress/theme'
 import { h } from 'vue'
 import './index.css'
-import Publications from '../components/Publications.vue'
+import {Publications} from '../components/Publications'
```
Comment (ice1000) [inline src/.vitepress/theme/index.ts:5]: Remove braces?

**Review by ice1000 — COMMENTED** [review]
How to create a new page similar to Publications/index.ts?

File: src/.vitepress/theme/index.ts (hunk 1 above)
Comment (Sepush) [reply src/.vitepress/theme/index.ts:5]: Publications is a named export so we need do this

**Comment by Sepush** [comment]
> How to create a new page similar to Publications/index.ts?

just like before or pass the data as prop for it
`<Publications :pubs="data">`

**Comment by ice1000** [comment]
> > How to create a new page similar to Publications/index.ts?
> 
> just like before or pass the data as prop for it
> `<Publications :pubs="data">`

Can you make `pubs.ts` use this syntax?

**Comment by Sepush** [comment]
> > > How to create a new page similar to Publications/index.ts?
> > 
> > 
> > just like before or pass the data as prop for it
> > `<Publications :pubs="data">`
> 
> Can you make `pubs.ts` use this syntax?

I made it be the default value of props pub so it's no need to pass a prop now

**Comment by ice1000** [comment]
How does pubs.ts work now? It doesn't use a `<Publications />` widget

**Comment by Sepush** [comment]
> How does pubs.ts work now? It doesn't use a `<Publications />` widget

- `pub.ts` just provide the data that match the type Publications I define in the interface.ts
- `Publications.tsx` this component import the data from `pub.ts` and pass it as the default value of props pub
- in the pubs/index.md we use `<Publications />` this is how it works

**Comment by ice1000** [comment]
Which one of these is pubs/index?

**Comment by ice1000** [comment]
I think you have to change it to use the data in pubs.ts am I right

**Comment by ice1000** [comment]
Can we have no default value but pass in argument explicitly?

**Comment by Sepush** [comment]
> I think you have to change it to use the data in pubs.ts am I right

I did not change this file so it not in the change files
the path is `/src/pubs`

**Comment by Sepush** [comment]
> Can we have no default value but pass in argument explicitly?

we can but have the default value is a better we 
so that we don't need import data and pass is in a .md file

**Comment by ice1000** [comment]
> > Can we have no default value but pass in argument explicitly?
> 
> we can but have the default value is a better we 
> so that we don't need import data and pass is in a .md file

But there is only one default value, and if I want to pass in data I'll have to do it myself. I'm just curious how to import data in .md

File: src/pubs/index.md [hunk 2]
```diff
@@ -2,4 +2,7 @@
 
 This is a list of publications about Aya by the Aya developers.
 
-<Publications />
+<script setup lang="ts">
+import {publications} from '/.vitepress/pubs'
+</script>
+<Publications :pubs="publications"/>
```
Comment (ice1000) [inline src/pubs/index.md:8]: Naruhoto

**Pull request merged**